eventstore/store.go

```diff
@@ -106,67 +119,71 @@ func (s *EventStore) Init() error {
 // Refresh triggers regeneration of the cached normalized view of the event dat and rebuilds the
 // billable components. Ideally you should do this once a day
 func (s *EventStore) Refresh() error {
+	// drop all the projection data
+	if err := s.refreshSQLFile("drop_ephemeral_objects.sql"); err != nil {
+		return err
+	}
```

Comment:
I wonder if there might be a bit of an issue here now we're splitting up the steps into little transactions.

Previously, we'd drop (for example) the `billable_event_components` table here, then recreate it, all in the same transaction. Now we're dropping it in one transaction (at which point I think the queries will start erroring), and then creating it again later (`create_billable_event_components.sql` further down).

I think you could reproduce this in a dev environment by putting a big sleep in the code here to recreate the situation we have in prod where `create_events.sql` takes several minutes.

I wonder if we might need to create temp tables and then move them for all the "ephemeral" tables, as they're probably all keyed off each other. 🤔